app/src/lib/http.ts

```diff
@@ -133,7 +133,7 @@ export async function parsedResponse<T>(response: Response): Promise<T> {
       throw new APIError(response, null)
     }
 
-    throw apiError
+    throw new APIError(response, apiError)
```

Comment:
Just nitpicking but could we move this up into the `try` block and thus get rid of the `apiError` let variable all together?

```ts
try {
  throw new APIError(response, await deserialize<IAPIError>(response))
} catch {
  throw new APIError(response, null)
}
```